ocfweb/docs/views/servers.py

```diff
@@ -57,95 +62,96 @@ def has_munin(self):
         return self.type in ('hypervisor', 'vm', 'server', 'desktop')
 
 
-@periodic(120)
+def is_hidden(host):
+    return host['cn'][0].startswith('hozer-') or host['cn'][0].startswith('dev-')
+
+
+PQL_GET_CHILDREN = "facts { name = 'vms' }"
+PQL_IS_HYPERVISOR = 'resources[certname] { type = "Class" and title = "Ocf_kvm" }'
+
+
+def query_puppet(query):
+    """Accepts a PQL query, returns a parsed json result"""
+    URL = 'https://puppetdb:8081/pdb/query/v4'
+    ROOT_DIR = '/etc/ocfweb/puppet-certs'
+    r = get(
+        URL,
+        cert=(join(ROOT_DIR, 'puppet-cert.pem'), join(ROOT_DIR, 'puppet-private.pem')),
+        verify=join(ROOT_DIR, 'puppet-ca.pem'),
+        params={'query': query},
+    )
+    return json.loads(r.text)
+
+
+def format_query_output(output):
+    """Converts the output of a puppet query to a dictionary of the form {certname: value}
+    """
+    result = {}
+    for d in output:
+        hostname = d['certname'].split('.')[0]
+        result[hostname] = d.get('value')
+    return result
+
+
+def create_hosts(lst):
+    """Accepts a list of raw ldap output, returns a dictionary of host
+    objects indexed by hostname
+    """
+    hosts = {}
+    for h in lst:
+        if not is_hidden(h):
+            description = h.get('description', [''])[0]
+            hostname = h['cn'][0]
+            hosts[hostname] = (Host(hostname, h['type'], description, ()))
+    return hosts
+
+
+def host_key(h):
```

Comment:
Rather than doing this, I'd just implement the __lt__(self, other_host) method in the Host class and then the sorting will work just fine without needing to specify a key on a sort.

Reply:
Hmm, I agree that this is a bit weird. Not sure if making a `__key` method is necessary, it might just be fine to make it a regular method (without the underscores). @ethanhs, thoughts on what best practice would be here?